File: src/scala/scala_echo/EchoServlet.scala
package scala_echo
import javax.servlet.http._
import java.util._
import java.text._

class EchoServlet extends HttpServlet {
  override def doGet(request: HttpServletRequest, response: HttpServletResponse) = 
  {
    val now = new Date
    val sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ")
    val format = "{ 'status': 0, 'timestamp': '%s', 'request-url': '%s', 'data': '%s' }"
    
    response.setContentType("application/json")
    response.getWriter.printf(format, sdf.format(now), request.getRequestURI, request.getParameter("echo"))
  }
}

File: src/java/java_echo/EchoServlet.java
package java_echo;

import java.io.IOException;
import javax.servlet.http.*;
import java.util.Date;
import java.text.SimpleDateFormat;

public class EchoServlet extends HttpServlet {
    public void doGet(HttpServletRequest req, HttpServletResponse resp)
            throws IOException {

        String jsonFormat = 
            "{ 'status': 0, 'timestamp': '%s', 'request-url': '%s', 'data': '%s' }";

        Date now = new Date();
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");

        resp.setContentType("application/json");
        resp.getWriter().printf(jsonFormat, sdf.format(now), req.getRequestURI(),
            req.getParameter("echo"));
    }
}